File: sol/solution/0001-0100/0006. Zigzag Conversion/Solution.scala
object Solution {
    def convert(s: String, numRows: Int): String = {
        if (numRows == 1 || s.length <= numRows) return s
        val rows = Array.fill(Math.min(numRows, s.length))("")
        var curRow = 0
        var goingDown = false
        for (c <- s) {
            rows(curRow) += c
            if (curRow == 0 || curRow == numRows - 1) goingDown = !goingDown
            curRow += (if (goingDown) 1 else -1)
        }
        rows.mkString("")
    }
}

File: sol/solution/0001-0100/0006. Zigzag Conversion/Solution.ts
function convert(s: string, numRows: number): string {
    if (numRows === 1 || s.length <= numRows) return s;
    let rows: string[] = Array(Math.min(numRows, s.length)).fill("");
    let curRow = 0;
    let goingDown = false;
    for (let c of s) {
        rows[curRow] += c;
        if (curRow === 0 || curRow === numRows - 1) goingDown = !goingDown;
        curRow += goingDown ? 1 : -1;
    }
    return rows.join("");
}